src/handlers.js

```diff
@@ -0,0 +1,59 @@
+const fs = require('fs');
+const path = require('path');
+const url = require('url');
+
+const { titles } = require('./data');
+
+const getRoot = (res) => {
+  const filePath = path.join(__dirname, '..', 'public', 'index.html');
+  fs.readFile(filePath, (err, file) => {
+    if (err) {
+      res.writeHead(500, { 'Content-Type': 'text/html' });
+      res.end('<h1>Sorry, Server Internal Error!!!</h1>');
+    } else {
+      res.writeHead(200, { 'Content-Type': 'text/html' });
+      res.write(file);
+      res.end();
+    }
+  });
+};
+
+const getFromPublic = (res, endPoint) => {
+  const extensionsObj = {
+    html: 'text/html',
+    css: 'text/css',
+    png: 'image/png',
+    js: 'application/javascript',
+  };
+  const extension = endPoint.split('.')[1];
+  const filePath = path.join(__dirname, '..', 'public', endPoint);
+  fs.readFile(filePath, (err, file) => {
+    if (err) {
+      res.writeHead(500, { 'Content-Type': 'text/html' });
+      res.end('<h1>Sorry, Server Internal Error!!!</h1>');
+    } else {
+      res.writeHead(200, { 'Content-Type': extensionsObj[extension] });
+      res.write(file);
+      res.end();
+    }
+  });
+};
+
+const getAutocomplete = (req, res, uri) => {
+  const queryData = url.parse(uri, true).query.data;
```

Comment:
you can get the endpoint again from the reqeust object, no need to pass it as an argument. `const url = req.url` or `const { url } = req`. 

Note:  `url. parse` and `querystring.parse` is deprecated least since last LTS version 14. To get the querystring you can do the following: 

```js
  const searchParams = new URLSearchParams(req.url.split('?')[1])
  console.log(searchParams.get('q')) // not this could be null so you might want to add a condition or a default value
```